tcpdf.php

```diff
@@ -18922,13 +18922,26 @@ protected function openHTMLTagHandler($dom, $key, $cell) {
 				$prevlinewidth = $this->GetLineWidth();
 				$this->SetLineWidth($hrHeight);
 
-				$lineStyle = array(
-					'color' => $tag['fgcolor'],
-					'cap'   => $tag['style']['cap'],
-					'join'  => $tag['style']['join'],
-					'dash'  => $tag['style']['dash'],
-					'phase' => $tag['style']['phase'],
-				);
+				$lineStyle = array();
+
+				if (isset($tag['fgcolor'])) {
+					$lineStyle['color'] = $tag['fgcolor'];
+				}
+
+				if (isset($tag['style']) && is_array($tag['style'])) {
+					if (isset($tag['style']['cap'])) {
+						$lineStyle['cap'] = $tag['style']['cap'];
+					}
+					if (isset($tag['style']['join'])) {
+						$lineStyle['join'] = $tag['style']['join'];
+					}
+					if (isset($tag['style']['dash'])) {
+						$lineStyle['dash'] = $tag['style']['dash'];
+					}
+					if (isset($tag['style']['phase'])) {
+						$lineStyle['phase'] = $tag['style']['phase'];
+					}
```

Comment:
Are `cap`, `join`, `dash`, and `phase` even valid CSS properties for an `<hr>` element? At least Firefox complains about them as invalid!

Reply:
Good catch, I would agree to say not but I am not the one that wrote this code 😄 (nor do I have a code example on how it is used)
Just fixing it

Reply:
Instead of using multiple if conditions you can use a for loop with a list of tags...

Reply:
Okay, rebasing and fixing with an for loop